pygpu/basic.py

```diff
@@ -44,6 +47,9 @@ def triu(A, inplace=True):
 def tril(A, inplace=True):
     if len(A.shape) != 2:
```

Comment:
Just use `A.nidm` instead of `len(A.shape)` here (same for triu), and all will be ok !